examples_tests/37.EXRSplit/main.cpp

```diff
@@ -0,0 +1,223 @@
+} PACK_STRUCT;
+
+struct ImageData
+{
+	ImageData(core::smart_refctd_ptr<asset::IAsset> _image) : asset(std::move(_image))
+	{
+		image_raw = core::smart_refctd_ptr_static_cast<asset::ICPUImage>(asset).get();
+	}
+
+	core::smart_refctd_ptr<asset::IAsset> asset;
+	asset::ICPUImage* image_raw;
+} PACK_STRUCT;
+#include "irr/irrunpack.h"
+
+const char* vertexSource = R"===(
+#version 430 core
+layout(location = 0) in vec4 vPos; //only a 3d position is passed from irrlicht, but last (the W) coordinate gets filled with default 1.0
+layout(location = 1) in vec4 vCol;
+
+//#include <irr/builtin/glsl/broken_driver_workarounds/amd.glsl>
+
+layout( push_constant, row_major ) uniform Block {
+	mat4 modelViewProj;
+} PushConstants;
+
+layout(location = 0) out vec4 Color; //per vertex output color, will be interpolated across the triangle
+
+void main()
+{
+    gl_Position = PushConstants.modelViewProj*vPos; //only thing preventing the shader from being core-compliant
+    Color = vCol;
+}
+)===";
+
+const char* fragmentSource = R"===(
+#version 430 core
+
+layout(location = 0) in vec4 Color; //per vertex output color, will be interpolated across the triangle
+
+layout(location = 0) out vec4 pixelColor;
+
+void main()
+{
+    pixelColor = Color;
+}
+)===";
+
+int main()
+{
+	// create device with full flexibility over creation parameters
+	// you can add more parameters if desired, check irr::SIrrlichtCreationParameters
+	irr::SIrrlichtCreationParameters params;
+	params.Bits = 24; //may have to set to 32bit for some platforms
+	params.ZBufferBits = 24; //we'd like 32bit here
+	params.DriverType = video::EDT_OPENGL; //! Only Well functioning driver, software renderer left for sake of 2D image drawing
+	params.WindowSize = dimension2d<uint32_t>(1280, 720);
+	params.Fullscreen = false;
+	params.Vsync = true; //! If supported by target platform
+	params.Doublebuffer = true;
+	params.Stencilbuffer = false; //! This will not even be a choice soon
+	auto device = createDeviceEx(params);
+
+	if (!device)
+		return 1; // could not create selected driver.
+
+	device->getCursorControl()->setVisible(false);
+
+	QToQuitEventReceiver receiver;
+	device->setEventReceiver(&receiver);
+
+	auto driver = device->getVideoDriver();
+	auto smgr = device->getSceneManager();
+	auto am = device->getAssetManager();
+
+	scene::ICameraSceneNode* camera = smgr->addCameraSceneNodeFPS(0, 100.0f, 0.001f);
+
+	camera->setPosition(core::vector3df(-4, 0, 0));
+	camera->setTarget(core::vector3df(0, 0, 0));
+	camera->setNearValue(0.01f);
+	camera->setFarValue(100.0f);
+
+	smgr->setActiveCamera(camera);
+
+	/// OpenEXR image data ///
+	asset::IAssetLoader::SAssetLoadParams lp;
+	auto image_bundle = am->getAsset("../../media/daily_pt_1.exr", lp);
+	assert(!image_bundle.isEmpty());
+
+	core::vector<ImageData> data;
+	for (auto i = 0ul; i < image_bundle.getSize(); ++i)
+		data.push_back(image_bundle.getContents().first[i]);
```

Comment:
now instead of doing any more crap with the images, just save them to EXR again, one by one